samcli/commands/delete/delete_context.py

```diff
@@ -77,32 +71,6 @@ def __enter__(self):
     def __exit__(self, *args):
         pass
 
-    def parse_config_file(self):
-        """
-        Read the provided config file if it exists and assign the options values.
-        """
-        config_provider = ConfigProvider(CONFIG_SECTION, [CONFIG_COMMAND])
-        config_options = config_provider(
-            config_path=self.config_file, config_env=self.config_env, cmd_names=[CONFIG_COMMAND]
-        )
-        if not config_options:
-            return
-
-        if not self.stack_name:
-            self.stack_name = config_options.get("stack_name", None)
-        # If the stack_name is same as the one present in samconfig file,
-        # get the information about parameters if not specified by user.
-        if self.stack_name and self.stack_name == config_options.get("stack_name", None):
```

Comment:
Do we still need to have this logic in the delete code somewhere? If I remember correctly, the config decorator will load all parameters into the click context.

So if the options in the config were all related to stack `xyz`, and I instead did `sam delete --stack-name abc`, will it use the parameters related to `xyz`, for stack `abc`?

The previous behaviour looks like it won't, and instead will just not set the other options listed below, but feel free to call this out if's a non issue

Reply:
This is a good catch. We may run integration tests manually to see if there are some tests which validates this behavior. 

It is also weird that we only set these parameters if stack name matches, if not we are just leaving that as it is, which might fail the command there 🤔

Reply:
I tried it out with my changes, and the behaviour now is that it will prioritize parameters passed in via the cli options over config values, but theres no logic to skip setting the other variables if the stack name differs between the config and cli parameter